apps/my.immich.app/routes/[...rest]/+page.ts

```diff
@@ -5,6 +5,16 @@ import { StorageKey } from '$lib';
 export const ssr = false;
 
 export const load = (async ({ url }) => {
+  if (url.searchParams.has('instanceUrl')) {
+    const instanceURL = url.searchParams.get('instanceUrl');
+
+    // remove the instanceURL for the redirect
+    url.searchParams.delete('instanceUrl');
+    const strippedParams = url.searchParams.toString();
+
+    redirect(302, `${instanceURL}${url.pathname}?${strippedParams}`);
```

Comment:
There is already logic for building the redirect URL below, shouldn't we just set instanceUrl from this param instead of using localstorage and then continue with the rest of the existing logic?

Reply:
Yeah, that sounds good. Probably something like:
```ts
let instanceUrl = localStorage.getItem(StorageKey.INSTANCE_URL) || '';
const instanceUrlOverride = url.searchParams.get('instanceUrl');
if(instanceUrlOverride) {
  instanceUrl = instanceUrlOverride;
  // delete from url
}

// rest as usual
```

Reply:
you might be able to just always delete it from the url, in which case you don't even need the if anymore, just
```ts
const instanceUrl = url.searchParams.get('instanceUrl') || localStorage.getItem(StorageKey.INSTANCE_URL) || '';
url.searchParams.delete('instanceUrl');
```